util/dockerfiles/devcontainer/Dockerfile

```
@@ -41,13 +42,33 @@
 # if the container is used for education or demonstration purposes.
 FROM --platform=${BUILDPLATFORM} ghcr.io/gem5/ubuntu-24.04_all-dependencies:latest as builder
 RUN git clone --branch v24.0 https://github.com/gem5/gem5
+
 WORKDIR /gem5
 RUN scons build/ALL/gem5.opt -j`nproc`
+RUN scons defconfig build/ALL_CHI build_opts/ALL && \
+    scons setconfig build/ALL_CHI RUBY_PROTOCOL_CHI=y && \
+    scons build/ALL_CHI/gem5.opt -j`nproc`
+RUN scons build/VEGA_X86/gem5.opt -j`nproc`
 
-# Stage 2: The final stage where we create the devcontainer image.
+# Stage 2: Create the devcontainer image.
 # This includes all dependencies for building and running gem5, the
 # dependencies for developing gem5, and a pre-built gem5 binary.
 FROM --platform=${BUILDPLATFORM} ghcr.io/gem5/ubuntu-24.04_all-dependencies:latest
+COPY --from=builder /gem5/build/ALL/gem5.opt /usr/local/bin/gem5-mesi
+COPY --from=builder /gem5/build/ALL_CHI/gem5.opt /usr/local/bin/gem5
```

Let's name this `gem5-chi` and then do a symlink from gem5 to gem5-chi

```suggestion
COPY --from=builder /gem5/build/ALL_CHI/gem5.opt /usr/local/bin/gem5-chi
RUN ln -s /usr/local/bin/gem5-chi /usr/local/bin/gem5
```